### ISSUE TITLE
The "tensorcores" option makes the model dumber.

### ISSUE THREAD

**Issue by BadisG** [oobabooga/text-generation-webui#5134]
### Describe the bug

Hello,

Using the "tensorcores" option changes the logits in a bad way. For example I asked this question to Mixtral:
*"When was born Cédric Doumbé? - Cédric Doumbé was born on August 30, 199..."*
The next token is supposed to be "2"
![6ypabsj](https://github.com/oobabooga/text-generation-webui/assets/110173477/b2d9efae-d93d-4ddb-8b6d-196088b139a2)

You can see that activating "tensorcores" objectively leads to a deterioration in the quality of the results (The only viable token is less likely to be chosen).

It's probably expected, the GPU calculations are prone to non determinism, but I think you should add a disclamer or something on such options, the description of "tensorcores" is currently this one:

*"Use llama-cpp-python compiled with tensor cores support. This increases performance on RTX cards. NVIDIA only.*"
This is the place to add a new sentence to warn users that using this option would decrease the overall precision or something.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Compare the logits with "tensorcores" activated and not.

### Logs

```shell
No logs.
```

### System Info

```shell
Windows 10
RTX 3060
```

**Comment by Ph0rk0z** [comment]
This is as weird as having different batch size changing the perplexity. I can't even go much above 512 to test that out since I am exclusively using GPU inference and it crashes.

For single batch I am getting nothing out of tensorcores either and never did. Unfortunately this is 100% a llama.cpp issue.
